tools/topsites.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import datetime
+import hashlib
+import hmac
+import os
+import requests
+import sys
+from urllib import urlencode
+from xml.dom.minidom import parseString
+
+from sqlalchemy import create_engine
+from sqlalchemy import Column
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from webcompat import app
+
+# Constant for alexa top site API
+ATS_ACTION_NAME = "TopSites"
+ATS_RESPONSE_GROUP_NAME = "Country"
+ATS_SERVICE_HOST = "ats.amazonaws.com"
+ATS_AWS_BASE_URL = "http://" + ATS_SERVICE_HOST + "/?"
+ATS_DATEFORMAT = "%Y-%m-%dT%H:%M:%S.%f"
+ATS_HASH_ALGORITHM = "HmacSHA256"
+ATS_COUNT = 100
+
+# Regions to dump to topsites.db
+REGIONS = ["GLOBAL", "US", "FR", "IN", "DE", "TW", "ID", "HK", "SG", "PL",
+           "GB", "RU"]
+
+# Alexa top site access and secret key
+ats_access_key = None
+ats_secret_key = None
+
+
+# Cache parsed sites, change priority if raised
+topsites = {}
+
+engine = create_engine("sqlite:///" + os.path.join(
+    app.config["BASE_DIR"], "topsites.db"))
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+
+class Site(Base):
+    """SQLAchemy base object for an Alexa top site."""
+    __tablename__ = "topsites"
+
+    url = Column(String, primary_key=True)
+    priority = Column(Integer)
+    country_code = Column(String)
+    ranking = Column(Integer)
+
+    def __init__(self, url, priority, country_code, ranking):
+        self.url = url
+        self.priority = priority
+        self.country_code = country_code
+        self.ranking = ranking
+
+Base.metadata.create_all(engine)
+
+
+def query_topsites(country_code, count=1000):
+    """Query top sites with given country code and count."""
+    for index in range(1, count, 100):
+        uri = build_uri(country_code, index)
+        response = requests.get(uri)
+
+        if response.status_code == 200:
```

Comment:
You get a `if` which is only testing success, but not the other cases. 
What should happen if there's something different than `response.status_code` 

* put the URI on retry queue?
* send a warning so that we know it's incomplete
* Something else?

Cf https://github.com/webcompat/webcompat.com/pull/1591#discussion_r121859159